crates/sp-domains/Cargo.toml

```diff
@@ -12,8 +12,10 @@ description = "Primitives of domains pallet"
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
+blake2 = { version = "0.10.5", default-features = false }
 merlin = { version = "2.0.1", default-features = false }
 parity-scale-codec = { version = "3.2.1", default-features = false, features = ["derive"] }
+rs_merkle = { version = "1.2.0", default-features = false }
```

Comment:
Hm... I didn't check the rest of the PR, but why are we pulling yet another merkle tree library into our codebase, is there something wrong with the one we use in other places (like `subspace-archiving`) so we don't need to compile even more?

Reply:
No problem with `merkle_light`, but feel `rs_merkle`'s API is somehow easier to use and it's advertised as the most advanced Rust merkle tree library (all these libraries should meet my needs though :P), and it has more downloads than merkle_light.

If I'm not mistaken, there are no other places using merkle tree right, and the code had been eliminated from `subspace-archiving`. I didn't notice `merkle_light` is still in the Cargo.toml of `subspace-archiving`, which is actually an unused dependency, I guess this is a mere leftover, should we remove it?

Reply:
I don't think we need any of advanced features, so I hope they don't cost much (from the code it seems like they could cost some RAM. Otherwise I don't mind it. Let's remove `merkle_light` from `subspace-archiving`, not sure how I forgot to get rid of it earlier.